src/app/organizations/organization/organization-starring/organization-starring.component.ts

```diff
@@ -0,0 +1,157 @@
+import { StarentryService } from '../../../shared/starentry.service';
+import { StarOrganizationService } from '../../../shared/star-organization.service';
+import { AlertService } from '../../../shared/alert/state/alert.service';
+import { first, takeUntil } from 'rxjs/operators';
+import { Observable, of as ObservableOf, Subject } from 'rxjs';
+import { OrganizationStarringService } from './organization-starring.service';
+
+@Component({
+  selector: 'app-organization-starring',
+  templateUrl: './organization-starring.component.html',
+  styleUrls: ['./organization-starring.component.scss']
+})
+export class OrganizationStarringComponent implements OnInit, OnDestroy, OnChanges {
+
+  @Input() organization: any;
+  @Output() change: EventEmitter<boolean> = new EventEmitter<boolean>();
+  private user: any;
+  public isLoggedIn: boolean;
+  public rate = false;
+  public total_stars = 0;
+  public disable = false;
+  private ngUnsubscribe: Subject<{}> = new Subject();
+  private starredUsers: User[];
+  constructor(private trackLoginService: TrackLoginService,
+              private userQuery: UserQuery,
+              private containerService: ContainerService,
+              private starringService: StarringService,
+              private starentryService: StarentryService,
+              private starOrganizationService: StarOrganizationService,
+              private organizationStarringService: OrganizationStarringService,
+              private alertService: AlertService) { }
+
+  ngOnInit() {
+    this.trackLoginService.isLoggedIn$.pipe(takeUntil(this.ngUnsubscribe)).subscribe(isLoggedIn => this.isLoggedIn = isLoggedIn);
+    this.userQuery.user$.pipe(takeUntil(this.ngUnsubscribe)).subscribe(user => {
+      this.user = user;
+      this.rate = this.calculateRate(this.starredUsers);
+    });
+  }
+
+  ngOnChanges() {
+    if (this.organization) {
+      this.setUpOrganization(this.organization);
+    }
+  }
+
+  ngOnDestroy() {
+    this.ngUnsubscribe.next();
+    this.ngUnsubscribe.complete();
+  }
+
+  setUpOrganization(organization: any) {
+    this.organization = organization;
+    this.getStarredUsers();
+  }
+
+  // Determine if user has starred organization.
+  calculateRate(starredUsers: User[]): boolean {
+    if (!this.user || !starredUsers) {
+      return false;
+    } else {
+      let matchingUser: User;
+      if (!starredUsers) {
+        return false;
+      }
+      matchingUser = starredUsers.find(user => user.id === this.user.id);
+      if (matchingUser) {
+        return true;
+      } else {
+        return false;
+      }
+    }
+  }
+
+  /**
+   * Handles the star button being clicked
+   *
+   * @memberof StarringComponent
+   */
+  setStarring() {
+    this.disable = true;
+    if (this.isLoggedIn) {
+
+      if (this.rate) {
+        const message = 'Unstarring ' + this.organization.name;
+        this.alertService.start(message);
+      } else {
+        const message = 'Starring ';
```

Comment:
Also, I'm not wild about having the same variable declared twice so close to each other; even though they are in their own blocks and this is legal. I would probably inline and not have the variable. And/or only invoke `this.alertService.start` only once.

Just my personal opinion, so feel free to ignore.